crates/uv-distribution/src/metadata/requires_dist.rs

```diff
@@ -127,10 +127,23 @@ impl RequiresDist {
                 .flatten()
                 .collect::<BTreeMap<_, _>>();
 
+            // Get additional settings
+            let empty_settings = ToolUvDependencyGroups::default();
+            let group_settings = project_workspace
+                .current_project()
+                .pyproject_toml()
+                .tool
+                .as_ref()
+                .and_then(|tool| tool.uv.as_ref())
+                .and_then(|uv| uv.dependency_groups.as_ref())
+                .unwrap_or(&empty_settings);
+
             // Flatten the dependency groups.
-            let mut dependency_groups =
-                FlatDependencyGroups::from_dependency_groups(&dependency_groups)
-                    .map_err(|err| err.with_dev_dependencies(dev_dependencies))?;
+            let mut dependency_groups = FlatDependencyGroups::from_dependency_groups(
+                &dependency_groups,
+                group_settings.inner(),
+            )
+            .map_err(|err| err.with_dev_dependencies(dev_dependencies))?;
 
             // Add the `dev` group, if `dev-dependencies` is defined.
             if let Some(dev_dependencies) = dev_dependencies {
```

Comment:
Note: the legacy special `tool.uv.dev-dependencies` group gets injected post-flattening. Currently this means we don't support `tool.uv.dependency-groups.dev.requires-python` affecting it. I think this is *fine* (you can just migrate to `dependency-groups.dev` if you want to apply requires-python to it).

Reply:
Can we error eagerly if you use both?

Reply:
Hmm... like "you're using Modern dependency-groups features, please migrate off Legacy dependency-groups?" even if there's no actual interaction with how they're using them?